### ISSUE TITLE
Add tests for usage of `CustomParticle` and `ParticleList` in `formulary` functions

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1865]
### Feature description

We should add tests for functions in `plasmapy.formulary` that decorated with `particle_input` that can accept `CustomParticle`-like and `ParticleList`-like arguments, in addition to the `Particle`-like arguments.

### Motivation

After the improvements to `particle_input` in #1057, functions decorated with `particle_input` can now nominally accept `CustomParticle`, `ParticleList`, and even some `Quantity` arguments for appropriately annotated parameters.  However, this new functionality is untested, so we can't be confident that it actually works.  Adding tests will give us confidence that things are working, and also help us figure out edge cases that need special handling (i.e. collisions functionality).

### Implementation strategy

This issue is dependent upon #341, in that the tests would only be applicable for functions that have been decorated with `particle_input`.  

For functions that have been decorated with `particle_input`, we should add tests for inputs that include:

 - `CustomParticle`
 - `Quantity` of physical type mass or charge, as appropriate 
 - `ParticleList`
 - `ParticleList`-like (i.e., a `list` like `["p+", "He-4 2+"]`)

### Additional context

The following are function in `plasmapy.formulary` that have particle inputs.  They should be checked off when they have the tests as described above.

 - [ ] Alfven_speed
 - [ ] Coulomb_logarithm
 - [ ] Hall_parameter
 - [ ] Knudsen_number
 - [ ] Lundquist_number
 - [ ] MaxwellianCollisionFrequencies
 - [ ] Maxwellian_1D 
 - [ ] Maxwellian_speed_1D
 - [ ] Maxwellian_speed_2D
 - [ ] Maxwellian_speed_3D
 - [ ] Maxwellian_velocity_2D
 - [ ] Maxwellian_velocity_3D
 - [ ] RelativisticBody
 - [ ] SingleParticleCollisionFrequencies
 - [ ] Spitzer_resistivity
 - [ ] cold_plasma_permittivity_LRP
 - [ ] cold_plasma_permittivity_SDP
 - [ ] collision_frequency
 - [ ] coupling_parameter
 - [ ] critical_density
 - [ ] deBroglie_wavelength
 - [ ] electron_thermal_conductivity
 - [ ] electron_viscosity
 - [ ] force_drift
 - [ ] fundamental_electron_collision_freq
 - [ ] fundamental_ion_collision_freq
 - [ ] gyrofrequency
 - [ ] gyroradius
 - [ ] impact_parameter
 - [ ] impact_parameter_perp
 - [ ] inertial_length
 - [ ] ion_sound_speed
 - [ ] ion_thermal_conductivity
 - [ ] ion_viscosity
 - [ ] kappa_thermal_speed
 - [ ] kappa_velocity_1D
 - [ ] kappa_velocity_3D
 - [ ] lower_hybrid_frequency
 - [ ] mass_density
 - [ ] mean_free_path
 - [ ] mobility
 - [ ] permittivity_1D_Maxwellian
 - [ ] plasma_frequency
 - [ ] relativistic_energy
 - [ ] resistivity
 - [ ] thermal_bremsstrahlung
 - [ ] thermal_speed
 - [ ] thermoelectric_conductivity

There's a chance that I'm missing a few or including a few extra.  If any come up that need special handling, then they should be crossed out here and addressed in another issue.

**Comment by StanczakDominik** [comment]
This is going to be fun to write if you parametrize over `lambda particle: function(particle_arg, **non_particle_args)`!

**Comment by namurphy** [comment]
> and also help us figure out edge cases that need special handling (i.e. collisions functionality).

Speaking of edge cases, see also #1867, #1871, #1872, and #1873. Yeesh! 👀 😅 😹
